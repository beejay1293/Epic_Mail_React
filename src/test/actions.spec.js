import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import nock from 'nock';

import {
  auth,
  messages,
  fetchSpecificMessage,
  changeDashboardState,
  signinSuccess,
  signupSuccess,
  signinError,
  signupError,
  inboxSuccess,
  sentSuccess,
  unreadSuccess,
  dashboardState,
  getSPecificMessage,
  authenticating
} from '../actions/index';
const middlewares = [ thunk ];
const mockStore = configureMockStore(middlewares);

describe('test async functions', () => {
  let store;
  beforeEach(() => {
    store = mockStore({})

  })
  afterEach(() => {
    nock.cleanAll()
  })

  it('it tests login method', () => {
    nock('https://andela-epic-mail.herokuapp.com').post('/api/v2/auth/login').reply(200, {})

    const email = 'mattimobolaji@gmail.com'
    const password = 'december'

    const userDetails = {
      email, password,
    };

    return store.dispatch(auth(userDetails)).then(() => {
      expect(store.getActions()).toMatchSnapshot()
    })
  })

  it('tests signup method', () => {
    nock('https://andela-epic-mail.herokuapp.com').post('/api/v2/auth/signup').reply(200, {})
    const userDetails = {
      firstname: 'mobolaji',
    lastname: 'matti',
    email: 'mAmmsmd@gmail.com',
    number: '64657895436',
    password: 'december',
    pw2: 'december'
    }

    return store.dispatch(auth(userDetails)).then(() => {
      expect(store.getActions()).toMatchSnapshot()
    })

  })

  it('tests get inbox messages endpoint', () => {
    nock('https://andela-epic-mail.herokuapp.com').get('/api/v2/messages').reply(200, {})

    return store.dispatch(messages()).then(() => {
      expect(store.getActions()).toMatchSnapshot();
    })
  })

  it('tests get sent messages endpoint', () => {
    nock('https://andela-epic-mail.herokuapp.com').get('/api/v2/messages/sent').reply(200, {})

    return store.dispatch(messages()).then(() => {
      expect(store.getActions()).toMatchSnapshot();
    })
  })

  it('tests get unread nmessages endpoint', () => {
    nock('https://andela-epic-mail.herokuapp.com').get('/api/v2/messages/unread').reply(200, {})

    return store.dispatch(messages()).then(() => {
      expect(store.getActions()).toMatchSnapshot();
    })
  })

  it('create signinsuccess action', () => {
    store.dispatch(signinSuccess({}))
      expect(store.getActions()).toMatchSnapshot();
  })

  it('tests getSpecific message endpoint', () => {
    nock('https://andela-epic-mail.herokuapp.com').get('/api/v2/messages/2').reply(200, {})

    return store.dispatch(fetchSpecificMessage('2')).then(() => {
      expect(store.getActions()).toMatchSnapshot();
    })
  })
  
  it('tests authenticating action', () => {
    store.dispatch(authenticating());
    expect(store.getActions()).toMatchSnapshot()
  })

  it('tests getSpecific message action', () => {
    store.dispatch(getSPecificMessage({}));
    expect(store.getActions()).toMatchSnapshot();
  })

  it('tests changeDashboardState action', () => {
    store.dispatch(changeDashboardState('sent'));
    expect(store.getActions()).toMatchSnapshot();
  })




})
