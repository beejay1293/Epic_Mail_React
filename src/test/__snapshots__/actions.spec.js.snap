// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test async functions create signinsuccess action 1`] = `
Array [
  Object {
    "payload": Object {},
    "type": "SIGNIN_SUCCESS",
  },
]
`;

exports[`test async functions it tests login method 1`] = `
Array [
  Object {
    "type": "AUTHENTICATING",
  },
  Object {
    "payload": undefined,
    "type": "SIGNUP_ERROR",
  },
]
`;

exports[`test async functions tests authenticating action 1`] = `
Array [
  Object {
    "type": "AUTHENTICATING",
  },
]
`;

exports[`test async functions tests changeDashboardState action 1`] = `
Array [
  Object {
    "body": "viewMessages",
    "payload": "sent",
    "type": "DASHBOARD_STATE",
  },
]
`;

exports[`test async functions tests get inbox messages endpoint 1`] = `
Array [
  Object {
    "type": "AUTHENTICATING",
  },
]
`;

exports[`test async functions tests get sent messages endpoint 1`] = `
Array [
  Object {
    "type": "AUTHENTICATING",
  },
]
`;

exports[`test async functions tests get unread nmessages endpoint 1`] = `
Array [
  Object {
    "type": "AUTHENTICATING",
  },
]
`;

exports[`test async functions tests getSpecific message action 1`] = `
Array [
  Object {
    "body": "readEmail",
    "payload": Object {},
    "type": "SPECIFIC_MESSAGE_SUCCESS",
  },
]
`;

exports[`test async functions tests getSpecific message endpoint 1`] = `
Array [
  Object {
    "type": "AUTHENTICATING",
  },
]
`;

exports[`test async functions tests signup method 1`] = `
Array [
  Object {
    "type": "AUTHENTICATING",
  },
  Object {
    "payload": undefined,
    "type": "SIGNUP_ERROR",
  },
]
`;
